.github/workflows/release.yaml

```diff
@@ -0,0 +1,32 @@
+name: Release
+
+on:
+  push:
+    branches: [main]
+    tags: ["v*"]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: true 
+      matrix:
+        os: [ubuntu-latest]
+        db_backend: [ goleveldb ]
+    env: # Or as an environment variable
+    steps:
+      - uses: actions/checkout@v3
+      - name: Get the version
+        id: get_version
+        run: echo ::set-output name=VERSION::${GITHUB_REF/refs\/tags\/v/}
+      - uses: actions/setup-go@v2
+        with:
+          go-version: "^1.17" # The Go version to download (if necessary) and use.
+      - name: Build
+        run: |
+          VERSION=${{ steps.get_version.outputs.VERSION }} DB_BACKEND=${{ matrix.db_backend }} make build
+          tar -czvf archwayd_${{ steps.get_version.outputs.VERSION }}_${{ matrix.os }}.tar.gz build/archwayd
+      - uses: actions/upload-artifact@v2
```

Comment:
The `actions/upload-artifact@v2` action is to upload artifacts to workflow runs so that the next step can download them. To add assets to a GitHub release, you should use [an action that interacts with the Releases API](https://github.com/softprops/action-gh-release#%EF%B8%8F-uploading-release-assets).